### DISCUSSION TITLE
ScatterPlotItem big matrix by axis ignoring masked (np.nan) values

### DISCUSSION THREAD

**Discussion by mesa-git** [pyqtgraph/pyqtgraph#2797] [Q&A] [answered]
Hey guys,

Basically i have a huge matrix that i want to plot as ScatterPlotItems
I am masking it in different ways and replacing masked values with np.nan

It looks like this:

![image001](https://github.com/pyqtgraph/pyqtgraph/assets/59180701/e463674a-2551-4385-813c-2fc08eb98f78)

Is there a fast way to ScatterPlot this?

First dimension would be the x-axis and the 2nd dimension would be the y-axis.

**Comment by pijyoi** [comment]
How are you currently plotting it now?

**Reply by mesa-git** [reply]
I'm not :)

The most obvious one would be to plot each point by for loops but that is extremely slow.

**Reply by pijyoi** [reply, marked as answer]
You could pass the output of the following to a `PlotDataItem` or `ScatterPlotItem`?
```python
x, y = np.nonzero(np.isfinite(arr))
```

**Reply by mesa-git** [reply]
seems to work in the sense that it returns the indices (x, y) of the values that are not np.nan

I still needed a for loop at the end to replace indices with values, but it works.

Thanks :)

**Reply by pijyoi** [reply]
```python
In [21]: mask = np.isfinite(arr)
In [22]: x, y = np.nonzero(mask)
In [23]: v = arr[x, y]
```
